Frontend/src/components/CommentModal.jsx

```diff
@@ -0,0 +1,69 @@
+import { useState, useEffect } from "react";
+import "./CommentModal.css"
+
+function CommentModal ({props, isOpen, onClose, gifUrl, cardId}) {
+    const [comments, setComments] = useState([]);
+    const [commentText, setCommentText] = useState("");
+
+    useEffect(() => {
+        if (isOpen) {
+            fetch(`http://localhost:3000/cards/${cardId}/comments`)
+                .then(response => response.json())
+                .then(data => setComments(data))
+                .catch(error => console.error('Error fetching comments:', error));
+        }
+    }, [isOpen, cardId]);
+
+    const handleInputChange = (e) => {
+        setCommentText(e.target.value)
+    }
+
+    const handleCommentPost = () => {
+        const newComment = {
+            text: commentText,
+            author: "Anonymous"
+        };
+        fetch(`http://localhost:3000/cards/${cardId}/comments`, {
+            method: 'POST',
+            headers: {
+                'Content-Type': 'application/json'
+            },
+            body: JSON.stringify(newComment)
+        })
+        .then(response => response.json())
+        .then(data => {
+            setComments([...comments, data]);
+            setCommentText("");
+        })
+        .catch(error => console.error('Error posting comment:', error));
+    }
+
+    // add onChange functions to each of these inputs
+    if(!isOpen) return null;
```

Comment:
recommended to have brackets even if it's one single if then return 

`if(!isOpen){
return null;
}`